### PR TITLE
Extend array connect to accept lists

### PR DESCRIPTION
**Pull request by hannahbos**
Previously the array parameters, had to be specified as numpy arrays. Now lists are supported as well, by internally converting inserted parameter lists to numpy arrays.

### CODE REVIEW

File: pynest/nest/lib/hl_api_connections.py [hunk 1]
```diff
@@ -442,11 +442,19 @@ def Connect(pre, post, conn_spec=None, syn_spec=None, model=None):
             sr("cvlit")
         elif isinstance(syn_spec, dict):
             for key,value in syn_spec.items():
-                if isinstance(value, (numpy.ndarray, numpy.generic)):
+
+                # if value is a list, it is converted to a numpy array
```
Comment (jougs) [inline pynest/nest/lib/hl_api_connections.py]: "is a list" -> "is a list or a tuple"

File: sli/fdstreamtest.cc [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+  if ( i.good() )
+    std::cerr << "Read/close: Good." << std::endl;
+  else
+  {
+    perror( NULL );
+    std::cerr << "Read/close: Error!" << std::endl;
+    return -1;
+  }
+}
```
Comment (jougs) [inline sli/fdstreamtest.cc]: This file was deleted some time ago and it got re-added by accident. Please remove it again.

**Comment by jougs** [comment]
I'm :+1: with respect to the changes that should actually be in the PR. I've updated the second comment, please have another look at it. Many thanks!

**Comment by jougs** [comment]
I did not mean to remove the files from the code base, but to remove the changes to them from the PR. Sorry for being imprecise.

**Comment by mschmidt87** [comment]
@hannahbos , please get this PR to pass all tests, so that the code reviewers can have a look.

**Comment by jougs** [comment]
@tammoippen could you please have a look and merge if you're happy? Thanks!

**Comment by tammoippen** [comment]
I am happy with the code. Nice work @hannahbos . :+1: and merging.

**Pull request merged**